### ISSUE TITLE
[Feature] Training Deep Networks without Learning Rates Through Coin Betting

### ISSUE THREAD

**Issue by kasungayan** [Microsoft/CNTK#2992]
Hi Team,

COntinuous COin Betting (COCOB) is a novel algorithm for stochastic subgradient descent (SGD) that does not require any learning rate setting, which was originally introduced in [1].  This would be very beneficial and handy for Deep learning tasks since this doesn't require any learning rate to tune. Also, it converges considerably faster than traditional momentum-based optimizers. 

Do you guys have any plans of integrating this feature to the CNTK stack in near future?

Thanks in advance.

[1] [COCOB Paper](https://arxiv.org/abs/1705.07795)

**Comment by KeDengMS** [comment]
Thanks for the suggestion, you may implement it using [universal learner](https://cntk.ai/pythondocs/cntk.learners.html#cntk.learners.universal). We also welcome community contributes to new features like this.

**Comment by kkc0923** [comment]
How about this?
https://github.com/souravsingh/nips-cocob-cntk

**Comment by kasungayan** [comment]
Thanks for sharing this @kkc0923 , probably this can be integrated to upcoming CNTK release  @KeDengMS ?

**Comment by KeDengMS** [comment]
This implementation uses UserLearner which may not be optimal in performance. It might be better to use [Universal Learner](https://cntk.ai/pythondocs/Manual_How_to_use_learners.html. We welcome third party pull request to have it under bindings/python/cntk/contrib/learners, as well as necessary tests.
